### ISSUE TITLE
Support new-style Illumina read pair IDs

### ISSUE THREAD

**Issue by koadman** [gjospin/PhyloSift#338]
The original Illumina FastQ format used /1 and /2 to indicate paired reads. Starting around 6-12 months ago with the introduction of casava 1.8, the pair identifiers changed so they are separated from the main read with a space and also may contain sample barcode info:

```
@EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG
```

The pair ID is the 1 in `1:Y:18:ATCACG`.

See the wikipedia page on FastQ for more details:

http://en.wikipedia.org/wiki/FASTQ_format

**Comment by gjospin** [comment]
The current system replaces the whole header line by an ID number to which we add /1 and /2.
Not sure if this is still needed.

**Comment by koadman** [comment]
Is it able to recognize paired reads and distinguish them from unpaired reads?

**Comment by gjospin** [comment]
No. We either read 1 read from each files or we read 2 consecutive reads from a single file and pair those.

**Comment by koadman** [comment]
I see. I recall a time where FastSearch.pm was checking that the headers of neighboring reads in the input files match, and would treat that set of reads as unpaired if not. That behavior supported interleaved FastQ where some reads had been filtered out due to e.g. quality or other issues. The code doesn't support that now and I wonder if this issue was added back when it did.

What to do with this issue? My thoughts in order of increasing awesomeness:
1) close it and forget about it
2) implement at least some error checking to ensure input reads are properly paired
3) (re)implement full support for filtered interleaved FastQ read pairs

**Comment by koadman** [comment]
^---- as much as i like error checking (1) might be good for this sprint, and save 2 or 3 for later?

**Comment by gjospin** [comment]
I agree, it would require quite a bit of testing and the quick thoughts I've had on the strategy ended in a particular scenario being hard/annoying to deal with.
I vote "save for later"

**Comment by gjospin** [comment]
Also, this fell in the "garbage in, garbage out" category in a previous discussion.

**Comment by koadman** [comment]
one thing that makes me feel better about postponing this is that incorrectly paired reads are exceedingly unlikely to hit the same marker in a metagenomic analysis. so they would effectively be treated as unpaired anyway. 16S amplicon data on the other hand...
